### DISCUSSION TITLE
Keeps on Spinning

### DISCUSSION THREAD

**Discussion by marco161polo161** [FreeTubeApp/FreeTube#5166] [Q&A]
I have issue with freetube installed on M1 MAX (64gb ram ) macbook pro  connected via 1gb cable to the Hyperoptic  router  with sync 900mbps up & 900mbps down.  Firstly it was 8k videos which plays for 1mins and then stops  and spins for ever, now even 2160p @60 keep spinning after 1 or so minutes, when it starts buffering it never stops.  With trying external media player - IIna media player    I get this error 
<img width="263" alt="Screenshot 2024-05-24 at 13 19 49" src="https://github.com/FreeTubeApp/FreeTube/assets/170729741/a0537466-a2f4-4ca5-8825-3248a882b916">

I have all the latest MacOS 14.5 

I really do not see reason why it does not work 
Regards 
Marco

**Comment by 4yman-0** [comment]
Maybe FreeTube tried to open MPV but failed because it does not exist?

**Reply by marco161polo161** [reply]
Thank you for your answer, but the problem is the same 
<img width="264" alt="Screenshot 2024-07-12 at 09 26 14" src="https://github.com/user-attachments/assets/ddc59e14-548f-4e20-b5f2-1a39ecbe3dab">

I do have the MPV installed
<img width="299" alt="Screenshot 2024-07-12 at 09 28 38" src="https://github.com/user-attachments/assets/bc649c53-b4a8-46a8-b3e3-921c88724583">

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Just making sure I understand correctly here. Is your question about IIna or the regular player within FreeTube?

**Reply by marco161polo161** [reply]
technically both  . Iina because of Error  and Freetube because of caching/buffering problem in dash or 4k/8K

**Comment by marco161polo161** [comment]
Let me explain  Firsty I am not able to open with iina because i got error as shown,  Secondly  Freetube behave very weirdly  first  minute or two of any videos are loaded pretty much immidately. However  after that  Freetube does  not seem to be buffrening any more  after that.  It does spin indefinately   

I have change the setting in download to just open in browser like thorium my default and it works without the a single buffer.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
So the iina issue is being tracked in https://github.com/FreeTubeApp/FreeTube/issues/4132 but we would like to analyze the 4k/8k issue you are seeing.

1. Launch FreeTube
2. Open dev console (ctrl + shift + i)
3. Watch a 8k/4k video
4. Let the video play untill the issue occurs
5. Check the console for red errors (the yellow ones are warnings and can be ignored)
6. Make a screenshot of the console
7. Upload here

**Reply by marco161polo161** [reply]
<img width="1920" alt="SCREEN" src="https://github.com/user-attachments/assets/ef52f8b0-db08-4369-b495-8b06c25dc576">
As requested please find attached screenshot of the consol  in freetube , This is made on windows.  Unfortunately control+shift+i does not seem to be working in Mac OS as my post  originated in mac

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
on Mac OS it would be `cmd + shift + i`

From the looks of it this isnt the full output of the console you need to scroll down in the console to see the rest of the errors. These yellow ones we can ignore

**Reply by marco161polo161** [reply]
<img width="716" alt="Second part" src="https://github.com/user-attachments/assets/e4b7eed3-9fe6-4d4e-b280-66c8260d8fef">
rest of the errors

**Reply by marco161polo161** [reply]
<img width="1799" alt="Screenshot 2024-07-17 at 11 43 12" src="https://github.com/user-attachments/assets/ce9a1bf0-367d-41bc-86d1-e49215601096">
This is the screenshot from the Mac os
Thank you for everything that you are doing for us Thank you for helping me sort issues.
Kind Regards
Marco

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
So if i understand these screenshots correctly you dont get any red errors when the issue occurs?

**Reply by marco161polo161** [reply]
If you say so , then yes, made a screenshot I do know very little about coding and errors.  Now i know how to look for them I will observe and keep you updated

**Reply by marco161polo161** [reply]
<img width="1797" alt="Screenshot 2024-07-17 at 12 24 48" src="https://github.com/user-attachments/assets/f2d45947-e062-4eb0-87cc-ae6a23d899cd">
Regarding those error,  I have been playing for about 5 minutes

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
How many times did the issue occur in those 5minutes?

**Reply by marco161polo161** [reply]
About 4 times

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Please check the network tab (you need to open it before starting the video, so that it can keep track of all of the requests), things that could cause problems are requests talking really long e.g. multiple seconds or one that errored (e.g. marked red or with a status code of 400 or higher).

**Comment by marco161polo161** [comment]
I have upload video to youtube  , I have also make a screen short of the read error/cancellation. 

[
<img width="551" alt="Screenshot 2024-07-18 at 12 17 55" src="https://github.com/user-attachments/assets/ec590920-a050-4fd6-b888-31bb907cc91d">
]

(https://www.youtube.com/watch?v=XiCsnN3matU)

Hope fully video is available to be watched

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Thank you for providing the video. At this moment there is nothing you can do about it to resolve this issue. We will try to reproduce this on our end to see what we can do about it.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
We have successfully identified the issue and we are happy to say that this wont be an issue (based on our testing) in the next minor release of FT.

**Reply by marco161polo161** [reply]
Thank you for your help. Thank you for excellent support resolving my problem and Thank for every thing that you are doing for the community. I am gland that i was able to provide usefull information. 
Regards
Marco

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
You're welcome :) 

Please dont hesitate to reach out again if this issue occurs after the next minor release.
